sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -348,28 +348,41 @@ case class InsertIntoTable(
     partition: Map[String, Option[String]],
     child: LogicalPlan,
     overwrite: Boolean,
-    ifNotExists: Boolean)
+    ifNotExists: Boolean,
+    options: Map[String, String])
```

Comment:
I just opened another PR that requires this map, #13206. That adds a hint for the number of writers per output partition so the optimizer can add the appropriate repartition and sort operation.